app/models/status_message.rb

```diff
@@ -111,10 +111,8 @@ def filter_mentions
     update(text: Diaspora::Mentionable.filter_people(text, people_allowed_to_be_mentioned))
   end
 
-  def subscribers
-    super.tap {|subscribers|
-      subscribers.concat(legitimate_mention_recipients)
-    }
+  def add_mention_subscribers?
+    super && author.local?
```

Comment:
Do we really need the `author.local?` here? When sending a StatusMessage the author is always local, and the subscribers of the StatusMessage are only added to the relayable-subscribers if the author is local.